### PR TITLE
doc/architecture.rst - edit up to "Cluster Map"

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/architecture.rst up to "Cluster Map", but not including "Cluster Map".

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -30,58 +30,52 @@ A Ceph Storage Cluster consists of multiple types of daemons:
...
-            +---------------+ +---------------+ +---------------+ +---------------+ 
-            |      OSDs     | |    Monitors   | |    Managers   | |      MDS      |
-            +---------------+ +---------------+ +---------------+ +---------------+ 
-
-A Ceph Monitor maintains a master copy of the cluster map. A cluster of Ceph
-monitors ensures high availability should a monitor daemon fail. Storage cluster
-clients retrieve a copy of the cluster map from the Ceph Monitor.
+A Ceph Monitor maintains a master copy of the cluster map. Having more than one
+Ceph monitor in the Ceph cluster ensures high availability in the event that
+one of the monitor daemons fails. The Ceph monitor provides copies of the cluster
```
Comment (anthonyeleven) [inline doc/architecture.rst:35]: Ceph Monitors maintain the master copy of the cluster map, which they provide to Ceph clients.  Provisioning multiple monitors within a cluster ensure availability in the event that one of the monitor daemons or its host fails.

File: doc/architecture.rst [hunk 2]
```diff
@@ -30,58 +30,52 @@ A Ceph Storage Cluster consists of multiple types of daemons:
...
-Storage cluster clients and each :term:`Ceph OSD Daemon` use the CRUSH algorithm
-to efficiently compute information about data location, instead of having to
-depend on a central lookup table. Ceph's high-level features include a
-native interface to the Ceph Storage Cluster via ``librados``, and a number of
-service interfaces built on top of ``librados``.
-
-
+Storage cluster clients and :term:`Ceph OSD Daemon`\s use the CRUSH algorithm
+to compute information about data location. This means that clients and OSDs do
+not have to depend on a central lookup table. Ceph's high-level features
```
Comment (anthonyeleven) [inline doc/architecture.rst:49]: OSDs are not bottlenecked by a central lookup table.

File: doc/architecture.rst [hunk 3]
```diff
@@ -30,58 +30,52 @@ A Ceph Storage Cluster consists of multiple types of daemons:
...
 comes through a :term:`Ceph Block Device`, :term:`Ceph Object Storage`, the
-:term:`Ceph File System` or a custom implementation you create using
-``librados``-- which is stored as RADOS objects. Each object is stored on an
-:term:`Object Storage Device`. Ceph OSD Daemons handle read, write, and
-replication operations on storage drives. With the default BlueStore back end,
-objects are stored in a monolithic database-like fashion.
+:term:`Ceph File System`, or a custom implementation that you create by using
+``librados``. The data received by the Ceph Storage Cluster is stored as RADOS
+objects. Each object is stored on an :term:`Object Storage Device` (this is
+also called an "OSD"). Ceph OSD Daemons control read, write, and replication
```
Comment (anthonyeleven) [inline doc/architecture.rst:61]: Ceph OSDs control

File: doc/architecture.rst [hunk 4]
```diff
@@ -30,58 +30,52 @@ A Ceph Storage Cluster consists of multiple types of daemons:
...
             Object         OSD          Drive
 
-Ceph OSD Daemons store data as objects in a flat namespace (e.g., no
-hierarchy of directories). An object has an identifier, binary data, and
-metadata consisting of a set of name/value pairs. The semantics are completely
-up to :term:`Ceph Client`\s. For example, CephFS uses metadata to store file
-attributes such as the file owner, created date, last modified date, and so
-forth.
+Ceph OSD Daemons store data as objects in a flat namespace (this means that
+objects are not stored in a hierarchy of directories). An object has an
```
Comment (anthonyeleven) [inline doc/architecture.rst:74]: namespace.  This means ...

File: doc/architecture.rst [hunk 5]
```diff
@@ -30,58 +30,52 @@ A Ceph Storage Cluster consists of multiple types of daemons:
...
+identifier, binary data, and metadata consisting of a set of name/value pairs.
```
Comment (anthonyeleven) [inline doc/architecture.rst:75]: s/a set of//

File: doc/architecture.rst [hunk 6]
```diff
@@ -103,17 +97,18 @@ forth.
...
-In traditional architectures, clients talk to a centralized component (e.g., a
-gateway, broker, API, facade, etc.), which acts as a single point of entry to a
-complex subsystem. This imposes a limit to both performance and scalability,
-while introducing a single point of failure (i.e., if the centralized component
-goes down, the whole system goes down, too).
+In traditional architectures, clients talk to a centralized component. This
+centralized component might be a gateway, a broker, an API, or a facade. A
+centralized component of this kind acts as a single point of entry to a complex
+subsystem. Architectures that rely upon this kind of centralized component
+incur a limit to performance and scalability and have a single point of
```
Comment (anthonyeleven) [inline doc/architecture.rst:104]: experience limits to

File: doc/architecture.rst [hunk 7]
```diff
@@ -103,17 +97,18 @@ forth.
...
+failure. If the centralized component goes down, the whole system goes down.
```
Comment (anthonyeleven) [inline doc/architecture.rst:105]: the whole system is unavailable.

File: doc/architecture.rst [hunk 8]
```diff
@@ -122,15 +117,15 @@ CRUSH Introduction
...
-to older approaches, and enables massive scale by cleanly distributing the work
-to all the clients and OSD daemons in the cluster. CRUSH uses intelligent data
-replication to ensure resiliency, which is better suited to hyper-scale storage.
-The following sections provide additional details on how CRUSH works. For a
-detailed discussion of CRUSH, see `CRUSH - Controlled, Scalable, Decentralized
-Placement of Replicated Data`_.
+Replication Under Scalable Hashing)` algorithm to compute information about
+object location instead of relying upon a central lookup table. CRUSH provides
+a better data management mechanism than do older approaches, and CRUSH enables
+massive scale by cleanly distributing the work to all the OSD daemons in the
```
Comment (anthonyeleven) [inline doc/architecture.rst:123]: s/cleanly//
to all OSDs in

File: doc/architecture.rst [hunk 9]
```diff
@@ -103,17 +97,18 @@ forth.
...
+incur a limit to performance and scalability and have a single point of
+failure. If the centralized component goes down, the whole system goes down.
 
-Ceph eliminates the centralized gateway to enable clients to interact with 
-Ceph OSD Daemons directly. Ceph OSD Daemons create object replicas on other
-Ceph Nodes to ensure data safety and high availability. Ceph also uses a cluster
-of monitors to ensure high availability. To eliminate centralization, Ceph 
-uses an algorithm called CRUSH.
+Ceph eliminates this centralized component. This enables clients to interact
+with Ceph OSD Daemons directly. Ceph OSD Daemons create object replicas on
```
Comment (anthonyeleven) [inline doc/architecture.rst:108]: We have the age-old "OSD" vs "OSD Daemons".  I'll always favor the former, but you're the boss.

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 7 above)
Comment (zdover23) [reply doc/architecture.rst:105]: That is better.

File: doc/architecture.rst (hunk 9 above)
Comment (zdover23) [reply doc/architecture.rst:108]: My preference has now been ratified by your preference. So: our preference will go into the docs.

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:35]: Accepted.

File: doc/architecture.rst (hunk 2 above)
Comment (zdover23) [reply doc/architecture.rst:49]: Accepted.

File: doc/architecture.rst (hunk 3 above)
Comment (zdover23) [reply doc/architecture.rst:61]: Accepted.

File: doc/architecture.rst (hunk 4 above)
Comment (zdover23) [reply doc/architecture.rst:74]: Oh, alright.

File: doc/architecture.rst (hunk 5 above)
Comment (zdover23) [reply doc/architecture.rst:75]: Accepted.

File: doc/architecture.rst (hunk 6 above)
Comment (zdover23) [reply doc/architecture.rst:104]: Accepted.

File: doc/architecture.rst (hunk 7 above)
Comment (zdover23) [reply doc/architecture.rst:105]: I've gone with "becomes unavailable" to emphasize the procedural cause-and-effect nature of the unavailability of the central component.

File: doc/architecture.rst (hunk 8 above)
Comment (zdover23) [reply doc/architecture.rst:123]: Accepted.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53366 - Reef backport
https://github.com/ceph/ceph/pull/53367 - Quincy backport

**Pull request merged**